src/infrastructure/Listener.ts

```diff
@@ -336,18 +316,76 @@ export class Listener implements IListener {
      */
-    public status(address: Address): Observable<TransactionStatusError> {
+    public status(address: Address, transactionHash?: string): Observable<TransactionStatusError> {
         this.subscribeTo(`status/${address.plain()}`);
         return this.messageSubject.asObservable().pipe(
             filter((_) => _.channelName === ListenerChannelName.status),
             filter((_) => _.message instanceof TransactionStatusError),
             map((_) => _.message as TransactionStatusError),
+            filter((_) => !transactionHash || _.hash.toUpperCase() == transactionHash.toUpperCase()),
             filter((_) => address.equals(_.address)),
         );
     }
 
+    /**
+     * Filters the transaction by hash if provided.
+     * @param transaction the transaction
+     * @param transactionHash the hash.
+     */
+    private filterHash(transaction: Transaction, transactionHash: string | undefined): boolean {
+        if (transactionHash === undefined) {
+            return true;
+        } else {
+            const metaHash = transaction.transactionInfo!.hash;
+            return metaHash !== undefined ? metaHash.toUpperCase() === transactionHash.toUpperCase() : false;
+        }
+    }
+
+    /**
+     * It filters a transaction by address using the aliases.
+     *
+     * This method delegates the rest loading as much as possible. It tries to filter by signer first.
+     *
+     * Note: this filter performs one extra rest call and it should be down in the pipeline.
+     *
+     * @param address the address.
+     * @return an observable filter.
+     */
+    private filterByNotifyAccount<T extends Transaction>(address: Address): OperatorFunction<T, T> {
+        return (transactionObservable): Observable<T> => {
+            return transactionObservable.pipe(
+                flatMap((transaction) => {
+                    if (this.isSigned(transaction, address)) {
+                        return of(transaction);
+                    }
+                    const namespaceIdsObservable = this.namespaceRepository.getAccountsNames([address]).pipe(
+                        map((names) => {
+                            return ([] as NamespaceName[])
+                                .concat(...Array.from(names.map((accountName) => accountName.names)))
+                                .map((name) => name.namespaceId);
+                        }),
+                    );
+                    return namespaceIdsObservable.pipe(
+                        filter((namespaceIds) => transaction.NotifyAccount(address, namespaceIds)),
+                        map(() => transaction),
+                    );
+                }),
+            );
+        };
+    }
+
+    /**
+     * Checks if the transaction is signer by an address.
+     * @param transition the transaction
+     * @param address the address.
+     */
+    private isSigned(transition: Transaction, address: Address): boolean {
```

Comment:
Does it make sense to move this function within the Transaction model?

Reply:
Yes, transaction NotifyAccount should also call this method